src/SMU-Keithley_4200-SCS/main.py

```diff
@@ -109,26 +138,84 @@ def __init__(self):
             "Limited 10 pA": 1e-11,
             "Limited 1 pA": 1e-12,
         }
-        
-    def find_ports(self):
 
-        if RUNNING_ON_4200SCS:
-            ports = ["LPTlib"]
-        else:
-            ports = ["LPTlib via xxx.xxx.xxx.xxx"]
+        self.speed_dict = {
+            "Very fast": 0.01,
+            "Fast": 0.1,
+            "Medium": 1.0,
+            "Slow": 10.0,
+        }
+
+        # Communication Parameter
+        self.port_string: str = "192.168.0.1"
+        self.identifier: str = "Keithley_4200-SCS_" + self.port_string
+        self.command_set: str = "LPTlib"
+        self.card_id: int = 1
+
+        self.lpt: lpt | Proxy | None = None
+        self.param: param | Proxy | None = None
+
+        # Measurement parameter
+        self.route_out: str = "Rear"
+        self.source: str = "Voltage in V"
+        self.protection: float = 100e-6
+        self.speed: str = "Very fast"
+        self.channel: str = "SMU1"
+
+        self.card_name = "SMU" + self.channel[-1]
+        self.pulse_channel = None
+
+        self.measured_voltage: float = 0.0
+        self.measured_current: float = 0.0
 
-        return ports
-        
-    def set_GUIparameter(self):
-        
-        GUIparameter = {
+        # Pulse Mode Parameters
+        self.pulse_master = False
+        self.pulse_mode: bool = False
+
+        self.pulse_count: int = 1
+        self.pulse_meas_start: float = 50
+        self.pulse_meas_duration: float = 20
+        self.pulse_width: float = 0.5e-6
+        self.pulse_period: float = 2e-6
+        self.pulse_delay: float = 1e-9
+        self.pulse_base_level: float = 0.0
+        self.pulse_rise_time: float = 100e-9
+        self.pulse_fall_time: float = 100e-9
+        self.pulse_impedance: float = 1e6
+
+        # List Mode Parameters
+        self.list_master: bool = False
+        """Only one channel can run the list sweep and is the master."""
+
+        self.list_receiver: bool = False
+        """If another channel runs a list sweep, the measurement must be done as list receiver."""
+
+        self.list_sweep_values: list[float] = []
+        self.list_measurement_keys = {
+            "voltage": "voltage",
+            "current": "current",
+            "time": "time",
+        }
+        """The keys are used to register and receive the list sweep values."""
+
+        self.list_delay_time: float = 0.0
+
+    @staticmethod
+    def find_ports() -> list[str]:
+        """Find available ports."""
+        return ["LPTlib"] if RUNNING_ON_4200SCS else ["LPTlib via xxx.xxx.xxx.xxx"]
```

Comment:
Is that what's shown to the user? Or is xxx.... somehow replaced with the actual IP Address?
More user friendly alternative would be "LPTlib via IP" or "LPTlib via TCPIP" or "LPTlib via network"

Reply:
The user has to exchange the xxx with the IP address of the server.exe